src/mscorlib/Common/System/SR.cs

```diff
@@ -0,0 +1,98 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Resources;
+using System.Runtime.CompilerServices;
+
+namespace System
+{
+    internal static partial class SR
+    {
+        private static ResourceManager s_resourceManager;
+
+        private static ResourceManager ResourceManager
+        {
+            get
+            {
+                if (SR.s_resourceManager == null)
+                {
```

Comment:
I suspect CoreCLR needs all of the existing protection in the Environment class since there's a lot more that can go wrong in various phases of CoreCLR startup.